src/Uno.UWP/Storage/PathIO.cs

```diff
@@ -0,0 +1,169 @@
+			AppendLinesTaskAsync(absolutePath, lines).AsAsyncAction();
+
+		public static IAsyncAction AppendLinesAsync(string absolutePath, IEnumerable<string> lines, UwpUnicodeEncoding encoding) =>
+			AppendLinesTaskAsync(absolutePath, lines, encoding).AsAsyncAction();
+
+		public static IAsyncOperation<IBuffer> ReadBufferAsync(string absolutePath) =>
+			ReadBufferTaskAsync(absolutePath).AsAsyncOperation();
+
+		public static IAsyncAction WriteBufferAsync(string absolutePath, IBuffer buffer) =>
+			WriteBufferTaskAsync(absolutePath, buffer).AsAsyncAction();
+
+		public static IAsyncAction WriteBytesAsync(string absolutePath, byte[] buffer) =>
+			WriteBytesTaskAsync(absolutePath, buffer).AsAsyncAction();
+
+		private static async Task<string> ReadTextTaskAsync(string path, UwpUnicodeEncoding? encoding = null)
+		{
+			if (path is null)
+			{
+				throw new ArgumentNullException(nameof(path));
+			}
+
+			var file = await StorageFile.GetFileFromPathAsync(path);
+			if (encoding == null)
+			{
+				return await FileIO.ReadTextAsync(file).AsTask();
+			}
+			return await FileIO.ReadTextAsync(file, encoding.Value).AsTask();
+		}
+
+		private static async Task<IList<string>> ReadLinesTaskAsync(string path, UwpUnicodeEncoding? encoding = null)
+		{
+			if (path is null)
+			{
+				throw new ArgumentNullException(nameof(path));
+			}
+
+			var file = await StorageFile.GetFileFromPathAsync(path);
+			if (encoding == null)
+			{
+				return await FileIO.ReadLinesAsync(file).AsTask();
+			}
+			return await FileIO.ReadLinesAsync(file, encoding.Value).AsTask();
+		}
+
+		private static async Task WriteTextTaskAsync(string path, string contents, bool append, UwpUnicodeEncoding? encoding = null)
+		{
+			if (path is null)
+			{
+				throw new ArgumentNullException(nameof(path));
+			}
+
+			var file = await StorageFile.GetFileFromPathAsync(path);
+			if (encoding == null)
+			{
+				return await FileIO.WriteTextAsync(file, contents, append).AsTask();
+			}
+			return await FileIO.WriteTextAsync(file, contents, append, encoding).AsTask();
+		}
+
+		private static async Task WriteBytesTaskAsync(IStorageFile file, byte[] buffer)
+		{
+			if (file is null)
+			{
+				throw new ArgumentNullException(nameof(file));
+			}
+
+			using var fs = File.Create(file.Path);
+			await fs.WriteAsync(buffer, 0, buffer.Length);
+		}
+
+		private static async Task<IBuffer> ReadBufferTaskAsync(IStorageFile file)
+		{
+			using var fs = File.OpenRead(file.Path);
+			var bytes = await fs.ReadBytesAsync();
+			return new UwpBuffer(bytes);
+		}
+
+		private static async Task WriteBufferTaskAsync(IStorageFile file, IBuffer buffer)
+		{
+			if (!(buffer is UwpBuffer inMemoryBuffer))
+			{
+				throw new NotSupportedException("The current implementation can only write a UwpBuffer");
+			}
+
+			await WriteBytesTaskAsync(file, inMemoryBuffer.Data);
+		}
+
+		private static async Task<Encoding> GetEncodingFromFileAsync(IStorageFile file)
```

Comment:
Looks like a duplicate of :

https://github.com/unoplatform/uno/blob/511e72d2d43b96911e4901e9de055e03c57bdaa6/src/Uno.UWP/Storage/FileIO.cs#L280

Reply:
Yes, it was a convenience copy paste initially, but I removed those and am using `FileIO` directly under the covers 👍